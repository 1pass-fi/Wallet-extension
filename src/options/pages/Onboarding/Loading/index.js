import React, { useEffect, useState } from 'react'
import loadingIcon from 'img/loading-icon.gif'

import './index.css'

export default ({ show }) => {
  const [showLoadingScreen, setShowLoadingScreen] = useState(false)

  useEffect(() => {
    const timeout = setTimeout(() => {
      setShowLoadingScreen(true)
    }, 300)
    return () => clearTimeout(timeout)
  }, [show])

  return !showLoadingScreen ? null : (
    <div className="create-wallet-loading-screen">
      <img className="create-wallet-loading-icon" src={loadingIcon} />
    </div>
  )
}
