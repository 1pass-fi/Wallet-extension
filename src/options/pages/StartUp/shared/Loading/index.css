.create-wallet-loading-screen {
  position: fixed;
  z-index: 500;
  background-color: rgba(255, 255, 255, 0.4);
  width: 100vw;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  user-select: none;
}

.create-wallet-loading-screen .create-wallet-loading-icon {
  width: 100px;
  height: 100px;
}
