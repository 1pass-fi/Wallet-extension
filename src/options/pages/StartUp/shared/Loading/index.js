import React from 'react'
import loadingIcon from 'img/loading-icon.gif'
import './index.css'

export default () => {
  return (
    <div className="create-wallet-loading-screen">
      <img className="create-wallet-loading-icon" src={loadingIcon} />
    </div>
  )
}
