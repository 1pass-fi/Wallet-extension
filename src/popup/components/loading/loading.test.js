import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import React from 'react'

import Loading from '.'

describe('Test for Loading component', () => {
  describe('Render without crashing', () => {
    it('renders correctly', () => {
      const { container } = render(<Loading />)
      expect(container).toMatchSnapshot()
    })
  })
})
