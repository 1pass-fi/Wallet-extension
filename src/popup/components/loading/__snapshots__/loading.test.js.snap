// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Loading component Render without crashing renders correctly 1`] = `
<div>
  <div
    class="loading-screen"
  >
    <img
      class="loading-icon"
      src="[object Object]"
    />
  </div>
</div>
`;
